app/components/SignInSignOut.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { Component, PropTypes } from 'react';
+import Spinner from 'react-mdl/lib/Spinner';
+import Constants from '../constants';
+import AuthActions from '../actions/auth';
+
+export default class SignInSignOut extends Component {
+  render() {
+    const { dispatch, auth } = this.props;
+    let content;
+
+    switch (auth.state) {
```

Comment:
So I guess this component can be removed completely and we'd need only sign out link there.